libvast/vast/system/signal_monitor.hpp

```diff
@@ -13,31 +13,19 @@
 
 #pragma once
 
+#include <atomic>
 #include <chrono>
 
-#include <caf/typed_actor.hpp>
-
-#include "vast/system/atoms.hpp"
+#include <caf/fwd.hpp>
 
 namespace vast::system {
 
-struct signal_monitor_state {
-  static inline const char* name = "signal-monitor";
-};
-
-using signal_monitor_type = caf::typed_actor<caf::reacts_to<run_atom>>;
+class signal_monitor {
+public:
+  static std::atomic<bool> stop;
 
-/// Monitors the application for UNIX signals.
-/// @note There must not exist more than one instance of this actor per
-///       process.
-/// @param self The actor handle.
-/// @param monitoring_interval The number of milliseconds to wait between
-///        checking whether a signal occurred.
-/// @param receiver The actor receiving the signals.
```

Comment:
Any reason not to keep the documentation? Since this class doesn't live in namespace `detail`, I think it's worthwhile having a short description in the header of how to use it.